### PR TITLE
Sroberge/perf picking

### PR DESCRIPTION
**Pull request by vim-sroberge**

### CODE REVIEW

File: src/inputMouse.ts [hunk 1]
```diff
@@ -104,22 +104,38 @@ export class InputMouse {
...
   onMouseClick = (position: Vector2) => {
-    /*
     console.time('raycast')
     const hits = this.mouseRaycast(position)
     console.timeEnd('raycast')
 
-    if (hits.length > 0) {
-      const mesh = hits[0].object
-      const index = hits[0].instanceId
+    if (!hits?.length) {
```
Comment (cdiggins) [inline src/inputMouse.ts:111]: Raycast test should be in a separate function from "onMouseClick"

File: src/VIMLoader.ts [hunk 2]
```diff
@@ -539,3 +452,108 @@ export class VIMLoader {
...
+  }
+
+  createAllGeometry = (): THREE.BufferGeometry[] => {
+    const meshCount = this.g3d.meshSubmeshes.length
+    const resultMeshes: THREE.BufferGeometry[] = []
+
+    for (let mesh = 0; mesh < meshCount; mesh++) {
+      const result = this.createBufferGeometryFromMeshIndex(mesh)
+      result?.computeBoundingSphere()
+      result?.computeBoundingBox()
```
Comment (cdiggins) [inline src/VIMLoader.ts:478]: Are you sure you need both?

Comment (vim-sroberge) [reply src/VIMLoader.ts:485]: I use sphere for centering the camera on object to provide better result for non aligned objects but maybe that's not required

**Review by mavimaec — APPROVED** [review]

**Review by cdiggins — APPROVED** [review]
Code looks good. Approving to unblock you. However I tried it on https://vimaec.github.io/vim-webgl-viewer/index-dev.html and I didn't see it working. Did you do a dev deploy of the package? I didn't see that at: https://www.npmjs.com/package/vim-webgl-viewer. It's a great way to show reviewers the new feature in action.

**Pull request merged**